applications/segment_everything/README.md

```diff
@@ -0,0 +1,134 @@
+
+
+Segment anything allows computing masks or pre-thresholded outputs for objects in natural images. An output mask is created for each query point. In this app the sam encoder and decoder parts are applied to a videostream.
+
+## Model
+
+This application uses a segment anything model (SAM) from https://github.com/facebookresearch/segment-anything/tree/main#model-checkpoints
+
+## Setup
+
+The setup requires three steps. 
+1.  create a pytorch docker container, with a tensorRT version that matches the holoscan tensorRT version. 
+2. Then this container creates the tensorRT engine files.
+3. We create a holoscan container, that mounts the created engine files, and the holohub repository, and runs the application. 
+
+## 1. pytorch container
+
+We need to use a pytorch container to create the TRT engine files for SAM models
+First get the docker: 
+```sh
+docker pull nvcr.io/nvidia/pytorch:23.03-py3
+``` 
+
+#### tensorRT version
+To use the trt engine files in holoscan later, we need to have the matching tensorrt version. 
+In this case we are looking for the version that matches 
+- os="ubuntu2004"
+- tag="8.6.1-cuda-12.0"
+- arm based system (using a CAGX devkit)
+
+following the instructions on https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-861/install-guide/index.html#downloading for the debian installation procedure.
+
+The tensorRT version needs to be downloaded from https://developer.nvidia.com/nvidia-tensorrt-8x-download
+
+In this case I download it to a folder on the m2 drive. e.g. 
+```sh
+mkdir /media/m2/software/tensorrt
+cd /media/m2/software/tensorrt
+
+wget https://developer.nvidia.com/downloads/compute/machine-learning/tensorrt/secure/8.6.1/local_repos/nv-tensorrt-local-repo-ubuntu2004-8.6.1-cuda-12.0_1.0-1_arm64.deb
+```
+
+#### run docker container
+Now the pytorch container can be run. Mount <HOLOHUB_SOURCE_DIR>, which contains the holohub repo, and mount the software folder, which contains the downloaded trt .deb installation file.
+
+```sh
+docker run --rm -it -v $PWD:/workspace -v /media/m2/software:/workspace/software nvcr.io/nvidia/pytorch:23.04-py3 /bin/bash
+```
+##### installation of tensorrt 
+follow the commandline steps below to install tensorRT and additional packages.
+```sh
+cd /workspace/software/tensorrt
+dpkg -i nv-tensorrt-local-repo-ubuntu2004-8.6.1-cuda-12.0_1.0-1_arm64.deb
+
+cp /var/nv-tensorrt-local-repo-ubuntu2004-8.6.1-cuda-12.0/nv-tensorrt-local-7148CA18-keyring.gpg /usr/share/keyrings/
+
+apt-get update
+
+apt-get install tensorrt
+apt-get install python3-libnvinfer-dev
+python3 -m pip install numpy onnx onnx-graphsurgeon
+```
+
+Check the installed version 
+```sh
+dpkg-query -W tensorrt
+```
+
+## 2. create trt engine files
+
+In the same docker container we can now proceed with building the trt engine files.
+First, pull the sam_trt_light repo, which contains a script to create the engine files.
+```
+cd /workspace
+git clone https://github.com/maximilianofir/sam_trt_light.git
```

Comment:
Can we just put the script directly in Holohub?

Reply:
It's more than a script, more like a package, which is why it's also install with pip install . 
I don't like the current solution of using a separate repo, but on the other hand, we may not want to have the dependencies of this second repo in holohub, e.g. pytorch, ... 

Happy to follow advice here, on what's best.